lua/rustaceanvim/lsp.lua

```diff
@@ -222,6 +231,23 @@ M.stop = function(bufnr)
   return clients
 end
 
+---Reload settings for the LSP client.
+---@param bufnr? number The buffer number, defaults to the current buffer
+---@return table[] clients A list of clients that will be have their settings reloaded
+M.reload_settings = function(bufnr)
+  bufnr = bufnr or vim.api.nvim_get_current_buf()
+  local clients = rust_analyzer.get_active_rustaceanvim_clients(bufnr)
+  ---@cast clients lsp.Client[]
+  for _, client in ipairs(clients) do
+    local settings = get_start_settings(vim.api.nvim_buf_get_name(bufnr), client.root_dir, config.server)
+    client.settings = settings
```

Comment:
We can disable this diagnostic, as it's a lua-language-server quirk.

```suggestion
    ---@diagnostic disable-next-line: inject-field
    client.settings = settings
```